L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc

```diff
@@ -0,0 +1,188 @@
+};
+
+L1EGammaEEProducer::L1EGammaEEProducer(const edm::ParameterSet &iConfig)
+    : multiclusters_token_(
+          consumes<l1t::HGCalMulticlusterBxCollection>(iConfig.getParameter<edm::InputTag>("Multiclusters"))),
+      calibrator_(iConfig.getParameter<edm::ParameterSet>("calibrationConfig")) {
+  produces<BXVector<l1t::EGamma>>("L1EGammaCollectionBXVWithCuts");
+}
+
+void L1EGammaEEProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  float minEt_ = 0;
+
+  std::unique_ptr<BXVector<l1t::EGamma>> l1EgammaBxCollection(new l1t::EGammaBxCollection);
+
+  // retrieve clusters 3D
+  edm::Handle<l1t::HGCalMulticlusterBxCollection> multiclusters_h;
+  iEvent.getByToken(multiclusters_token_, multiclusters_h);
+  const l1t::HGCalMulticlusterBxCollection &multiclusters = *multiclusters_h;
+
+  std::vector<const l1t::HGCalMulticluster *> selected_multiclusters;
+  std::map<std::pair<int, int>, std::vector<const l1t::HGCalMulticluster *>> etaphi_bins;
+
+  // here we loop on the TPGs
+  for (auto cl3d = multiclusters.begin(0); cl3d != multiclusters.end(0); cl3d++) {
+    // std::cout << "-- CL3D is EG: " <<  cl3d->hwQual() << "   "<< cl3d->eta() <<"   "<< cl3d->pt()<<std::endl;
+    if (cl3d->hwQual()) {
+      if (cl3d->et() > minEt_) {
+        int hw_quality = 1;  // baseline EG ID passed
+        if (std::abs(cl3d->eta()) >= 1.52) {
+          hw_quality = 2;  // baseline EG ID passed + cleanup of transition region
+        }
+
+        float calib_factor = calibrator_.calibrationFactor(cl3d->pt(), cl3d->eta());
+        l1t::EGamma eg =
+            l1t::EGamma(reco::Candidate::PolarLorentzVector(cl3d->pt() / calib_factor, cl3d->eta(), cl3d->phi(), 0.));
+        eg.setHwQual(hw_quality);
+        eg.setHwIso(1);
+        eg.setIsoEt(-1);  // just temporarily as a dummy value
+        l1EgammaBxCollection->push_back(0, eg);
+        if (hw_quality == 2) {
+          // we build the EM interpreted EG object
+          l1t::EGamma eg_emint = l1t::EGamma(reco::Candidate::PolarLorentzVector(
+              cl3d->iPt(l1t::HGCalMulticluster::EnergyInterpretation::EM), cl3d->eta(), cl3d->phi(), 0.));
+          eg_emint.setHwQual(4);
+          eg_emint.setHwIso(1);
+          eg_emint.setIsoEt(-1);  // just temporarily as a dummy value
+          l1EgammaBxCollection->push_back(0, eg_emint);
+          // we also prepare for the brem recovery procedure
+          selected_multiclusters.push_back(&(*cl3d));
+          auto eta_phi_bin = get_eta_phi_bin(&(*cl3d));
+          auto bucket = etaphi_bins.find(eta_phi_bin);
+          if (bucket == etaphi_bins.end()) {
+            std::vector<const l1t::HGCalMulticluster *> vec;
+            vec.push_back(&(*cl3d));
+            etaphi_bins[eta_phi_bin] = vec;
+          } else {
+            bucket->second.push_back(&(*cl3d));
+          }
+        }
+      }
+    }
+  }
+
+  // std::cout << "# of selected HGCAL multiclusters: " << selected_multiclusters.size() << std::endl;
```

Comment:
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.